cmd/telegraf/telegraf.go

```diff
@@ -161,30 +163,31 @@ func (t *Telegraf) watchLocalConfig(signals chan os.Signal, fConfig string) {
 }
 
 func (t *Telegraf) runAgent(ctx context.Context) error {
-	// If no other options are specified, load the config file and run.
-	c := config.NewConfig()
-	c.OutputFilters = t.outputFilters
-	c.InputFilters = t.inputFilters
-	var err error
+	var configFiles []string
 	// providing no "config" flag should load default config
 	if len(t.config) == 0 {
-		err = c.LoadConfig("")
-		if err != nil {
-			return err
-		}
-	}
-	for _, fConfig := range t.config {
-		err = c.LoadConfig(fConfig)
-		if err != nil {
-			return err
-		}
+		configFiles = append(configFiles, "")
+	} else {
+		configFiles = append(configFiles, t.config...)
 	}
 
 	for _, fConfigDirectory := range t.configDir {
-		err = c.LoadDirectory(fConfigDirectory)
+		files, err := config.WalkDirectory(fConfigDirectory)
 		if err != nil {
 			return err
 		}
+		configFiles = append(configFiles, files...)
+	}
+
+	// If no other options are specified, load the config file and run.
+	c := config.NewConfig()
+	c.OutputFilters = t.outputFilters
+	c.InputFilters = t.inputFilters
+	var err error
```

Comment:
```suggestion
```
This `err` seems unused now since it was moved

Reply:
It's still used, moved initialization lower